packages/search/src/build-snippet-index.ts

```diff
@@ -16,14 +17,16 @@ type Context = {
 };
 
 async function indexFile(context: Context, file: File) {
-  const fileContent = await context.contentReader(file.filePath);
+  const filePath = isAbsolute(file.filePath) ? file.filePath : join(file.directory, file.filePath);
```

Comment:
In most cases, the  file path was already absolute, so this bug didn't manifest. But in a test case that passed a relative path for the file path, this bug was revealed. It should do the right thing in either case.